src/core/Object3D.js

```diff
@@ -534,44 +547,104 @@ THREE.Object3D.prototype = {
 
 	},
 
-	updateMatrix: function () {
-
-		this.matrix.compose( this.position, this.quaternion, this.scale );
-
-		this.matrixWorldNeedsUpdate = true;
-
-	},
-
-	updateMatrixWorld: function ( force ) {
+	updateMatrix: (function () {
+		
+	
+		var checkLastStateChanged = function(object3D){
+			
+			var position = object3D.position;
+			var quaternion = object3D.quaternion;
+			var scale = object3D.scale;
+			var matrixState = object3D._matrixState;
+			
+			return matrixState[0] !== position.x ||
+				matrixState[1] !== position.y ||
+				matrixState[2] !== position.z ||
+				matrixState[3] !== quaternion.x ||
+				matrixState[4] !== quaternion.y ||
+				matrixState[5] !== quaternion.z ||
+				matrixState[6] !== quaternion.w ||
+				matrixState[7] !== scale.x ||
+				matrixState[8] !== scale.y ||
+				matrixState[9] !== scale.z
+				
+		};
+		var setLastState = function(object3D){
+			
+			var position = object3D.position;
+			var quaternion = object3D.quaternion;
+			var scale = object3D.scale;
+			var matrixState = object3D._matrixState;
+			
+			matrixState[0] = position.x;
+			matrixState[1] = position.y;
+			matrixState[2] = position.z;
+			matrixState[3] = quaternion.x;
+			matrixState[4] = quaternion.y;
+			matrixState[5] = quaternion.z;
+			matrixState[6] = quaternion.w;
+			matrixState[7] = scale.x;
+			matrixState[8] = scale.y;
+			matrixState[9] = scale.z;
+				
+		};
+	
+		return function(){		
+			if 	(checkLastStateChanged(this)){
+				this.matrixCached.compose( this.position, this.quaternion, this.scale );
+				this.matrixWorldNeedsUpdate = true
```

Comment:
The problem with that is that when calling `updateMatrixworld` on a child object and the position of the parent is changed (without calling `updateMatrix`) then it wouldn't update the `matrixWorld` automatically for the child because it wouldn't see the that something has changed.